add page for aws service catalog integration (#507)
hashicorp/terraform-docs-common: website/docs/cloud-docs/integrations/aws-service-catalog/index.mdx

---

> page_title: Terraform Cloud for AWS Service Catalog

Reviewer: Question: Should we update this title to mirror the nav title [here](https://github.com/hashicorp/terraform-docs-common/pull/507/files#diff-db6bf0196608ec8d6db237317e526dca37469066eecb51be266577ac0dbcd554R583)? I recognize that this is the name of the GitHub repo, but it might be nice to generalize it in the docs a bit since it's for both TFC/E.
```suggestion
page_title: AWS Service Catalog Integration
```

Author: I think the current name matches the example set by the other integrations a little better. I'd have no objections to changing this title in a future PR though

description: >-

>   Provision infrastructure with HashiCorp Terraform Cloud inside AWS Service Catalog

Reviewer: Quick question / suggestion: Should we generalize this a bit since it's technically doable in both TFC and TFE?
```suggestion
  Provision infrastructure with HashiCorp Terraform Cloud or Terraform Enterprise inside AWS Service Catalog
```

Author: https://github.com/hashicorp/terraform-docs-common/pull/507#discussion_r1414335429

---

> # Terraform Cloud for AWS Service Catalog overview

Reviewer: Same question / suggestion as above: Should we mirror the nav's title here and generalize this a bit? Also, (nitpick) I believe that the "o" in "overview" should be capitalized here for consistency:
```suggestion
# AWS Service Catalog Integration Overview
```

Author: I did this because I was actually following this page as an example: https://developer.hashicorp.com/terraform/cloud-docs/integrations/kubernetes

> This integration allows administrators to curate a portfolio of pre-approved Terraform configurations on AWS Service Catalog. This enables end users like engineers, database administrators, and data scientists to deploy these Terraform configurations with a single action from the AWS interface. By combining Terraform Cloud with AWS Service Catalog, we’re combining a self-service interface that many customers are familiar with, AWS Service Catalog, with the existing workflows and policy guardrails of Terraform Cloud.

Reviewer: Similar question / suggestion as above: Should we include TFE?:
```suggestion
This integration allows administrators to curate a portfolio of pre-approved Terraform configurations on AWS Service Catalog. This enables end users like engineers, database administrators, and data scientists to deploy these Terraform configurations with a single action from the AWS interface. By combining Terraform Cloud and Terraform Enterprise with AWS Service Catalog, we’re combining a self-service interface that many customers are familiar with, AWS Service Catalog, with the existing workflows and policy guardrails of Terraform Cloud and Terraform Enterprise.
```

<!-- BEGIN: TFC:only name:pnp-callout -->
@include 'tfc-package-callouts/aws-service-catalog.mdx'
<!-- END: TFC:only name:pnp-callout -->

## Installation & Configuration

To start using this integration, you'll need to install the [AWS Service Catalog Engine for Terraform Cloud](https://github.com/hashicorp/aws-service-catalog-engine-for-tfc) provided by HashiCorp on GitHub by following the [setup instructions](https://github.com/hashicorp/aws-service-catalog-engine-for-tfc#getting-started) provided in the README. If you run into any setup troubles along the way, the README also includes [troubleshooting steps](https://github.com/hashicorp/aws-service-catalog-engine-for-tfc#troubleshooting) that should help resolve common issues that you may encounter.

> With the engine installed, the necessary code and infrastructure to integrate the Terraform Cloud engine with AWS Service Catalog will automatically be configured. The setup can be completed in just a few minutes, and it only needs to be done once. Once the setup is complete, you can immediately start using AWS Service Catalog to develop and manage AWS Service Catalog products, and make them accessible to your end users across all your accounts.

Reviewer: Similar suggestion as above: Should we generalize this a bit?:
```suggestion
With the engine installed, the necessary code and infrastructure to integrate the Terraform engine of your choice with AWS Service Catalog will automatically be configured. The setup can be completed in just a few minutes, and it only needs to be done once. Once the setup is complete, you can immediately start using AWS Service Catalog to develop and manage AWS Service Catalog products, and make them accessible to your end users across all your accounts.
```

Author: https://github.com/hashicorp/terraform-docs-common/pull/507#discussion_r1414335429

## Usage

> You can access this new feature through the AWS Service Catalog console in any AWS regions where AWS Service Catalog is supported and follow the AWS' Service Catalog Administrator Guide to [create your first Terraform Cloud product](https://docs.aws.amazon.com/servicecatalog/latest/adminguide/getstarted-product-Terraform.html).

Author: I'd prefer it matches the AWS docs and the repository name.

> You can access this new feature through the AWS Service Catalog console in any AWS regions where AWS Service Catalog is supported and follow the AWS' Service Catalog Administrator Guide to [create your first Terraform Cloud product](https://docs.aws.amazon.com/servicecatalog/latest/adminguide/getstarted-product-Terraform.html).

Reviewer: Got it. We do mention TFE in the repo readme, though, so maybe worth circling back around to in the future to ensure the two are consistent.
